scripts/datacube.py

```diff
@@ -147,25 +174,15 @@ def search_sentinel2(week, aoi, cloud_cover_percentage, nodata_pixel_percentage)
 
     s2_items_gdf = gpd.GeoDataFrame.from_features(s2_items.to_dict())
 
-    best_nodata = (
-        s2_items_gdf[["s2:nodata_pixel_percentage"]]
-        .groupby(["s2:nodata_pixel_percentage"])
-        .sum()
-        .sort_values(by="s2:nodata_pixel_percentage", ascending=True)
-        .index[0]
-    )
-
-    best_clouds = (
-        s2_items_gdf[["eo:cloud_cover"]]
-        .groupby(["eo:cloud_cover"])
-        .sum()
-        .sort_values(by="eo:cloud_cover", ascending=True)
-        .index[0]
-    )
+    least_nodata_and_clouds = s2_items_gdf.sort_values(
+        by=["s2:nodata_pixel_percentage", "eo:cloud_cover"], ascending=True
+    ).index[0]
```

Comment:
@lillythomas, the linter was picking up that the `best_nodata` variable wasn't used. I've refactored this to sort based on lowest `s2:nodata_pixel_percentage` first, and then lowest `eo:cloud_cover`, so the resulting dataframe (searching between dates `2020-01-13/2020-01-19`) would look something like this:

```
                      datetime  s2:nodata_pixel_percentage  eo:cloud_cover
6  2020-01-17T18:47:09.024000Z                    0.000000       33.132727
5  2020-01-22T18:46:51.024000Z                    0.000017       40.740009
1  2020-02-11T18:45:11.024000Z                    0.000043        2.595613
0  2020-02-16T18:44:39.024000Z                    0.000056       10.043734
4  2020-01-27T18:46:29.024000Z                    0.000066       12.765600
2  2020-02-06T18:45:39.024000Z                    0.000136        2.524842
3  2020-02-01T18:46:11.024000Z                    0.000169        1.933268
```

And we would pick the first row here (datetime: `2020-01-17T18:47:09.024000Z`). Does this look ok? Or should we change the logic a bit (e.g. as long as the nodata_pixel_percentage is <20%, we just pick the lowest cloud cover, which would then be datetime: `2020-02-01T18:46:11.024000Z`)

Reply:
Yes, I knew that `best_nodata` wasn't being used because I was mainly thinking that optimizing for least cloud cover is our best bet given that there is a pre-existing filter for nodata by way of `{"op": "<=", "args": [{"property": "s2:nodata_pixel_percentage"}, nodata_pixel_percentage]}` in the search query. I think this is a good alternative, but in some/many cases it won't optimize for cloud cover (as this example shows where the first row has a nontrivial amount of cloudy pixels - 33 %).

Reply:
Ok, so we would just pick the least cloudy image then, as long as the `nodata_pixel_percentage` is below the 20% threshold? I.e. simply remove the `best_nodata` variable and stick to having just `best_clouds`?

Reply:
Yes I think the best call is for us to just remove the `best_nodata`  variable 👍

Reply:
Done at f65e34a9c4f72950b4f8a68a6ea2d4c580878ecd!